domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotterTest.kt

```diff
@@ -231,6 +240,127 @@ class CpuPerformanceSnapshotterTest {
     assertThat(fakePerformanceMetricsEventLogger.noPerformanceMetricsEventsPresent()).isTrue()
   }
 
+  @Test
+  fun testSnapshotter_onCreate_moveToForegroundBeforeCutOff_logsCpuUsageInForegroundAfterDelay() {
+    applicationLifecycleObserver.onCreate()
+    testCoroutineDispatchers.runCurrent()
+    // clearing up all app startup performance metrics: apk size and storage usage.
+    fakePerformanceMetricsEventLogger.clearAllPerformanceMetricsEvents()
+    testCoroutineDispatchers.advanceTimeBy(1000)
+    applicationLifecycleObserver.onAppInForeground()
+    testCoroutineDispatchers.runCurrent()
+    testCoroutineDispatchers.advanceTimeBy(foregroundCpuLoggingTimePeriodMillis)
+
+    val count = fakePerformanceMetricsEventLogger.getPerformanceMetricsEventListCount()
+    val event = fakePerformanceMetricsEventLogger.getMostRecentPerformanceMetricsEvent()
+
+    assertThat(count).isEqualTo(1)
+    assertThat(event).isNotNull()
```

Comment:
Can this even be null? It seems like ``getMostRecentPerformanceMetricsEvent`` can't return null.

Reply:
nice catch -- removed.